common/dynamicconfig/constants.go

```diff
@@ -884,9 +884,12 @@ const (
 	HistoryTaskDLQEnabled = "history.TaskDLQEnabled"
 	// HistoryTaskDLQUnexpectedErrorAttempts is the number of task execution attempts before sending the task to DLQ.
 	HistoryTaskDLQUnexpectedErrorAttempts = "history.TaskDLQUnexpectedErrorAttempts"
-	// HistoryTaskDLQInteralErrors causes history task processing to send tasks failing with serviceerror.Internal to
+	// HistoryTaskDLQInternalErrors causes history task processing to send tasks failing with serviceerror.Internal to
 	// the dlq (or will drop them if not enabled)
 	HistoryTaskDLQInternalErrors = "history.TaskDLQInternalErrors"
+	// HistoryTaskDLQErrorSubStrings specifies a comma separated list of strings. If a task processing error contains
```

Comment:
I think some of our error messages do contain comma.

Reply:
Why not take a regular expression instead of substrings? That seems simpler (though YAML may make that difficult, I'm not sure)

Reply:
Regular expressions would be easier to compare too, just run `regexp.MatchString(e.dlqErrorPattern(), err.Error())`

Reply:
I tried to keep it simple this way. But yeah, regular expressions make sense.

Reply:
Definitely use a regex (then add tests for it ofc). They're more powerful, will take less code on our end to handle, and will execute faster than the "list of substrings" approach.

Reply:
Updated code to use Regex.